File: nestjs-api/src/controllers/analytics.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { AnalyticsService } from '../services/analytics.service';

@Controller('api/v1')
export class AnalyticsController {
  constructor(private readonly analyticsService: AnalyticsService) {}

  @Get('orders-with-users')
  async getOrdersWithUsers(@Query('limit') limit: string = '10', @Query('offset') offset: string = '0') {
    return this.analyticsService.getOrdersWithUsers(parseInt(limit), parseInt(offset));
  }

  @Get('user-order-summary')
  async getUserOrderSummary(@Query('limit') limit: string = '10', @Query('offset') offset: string = '0') {
    return this.analyticsService.getUserOrderSummary(parseInt(limit), parseInt(offset));
  }

  @Get('analytics')
  async getComplexAnalytics() {
    return this.analyticsService.getComplexAnalytics();
  }

  @Get('health')
  async healthCheck() {
    return {
      status: 'healthy',
      timestamp: new Date(),
      service: 'nestjs-api',
    };
  }
} 

File: dotnet-api/Controllers/AnalyticsController.cs
using Microsoft.AspNetCore.Mvc;
using DotNetApi.Services;

namespace DotNetApi.Controllers
{
    [ApiController]
    [Route("api/v1")]
    public class AnalyticsController : ControllerBase
    {
        private readonly AnalyticsService _analyticsService;

        public AnalyticsController(AnalyticsService analyticsService)
        {
            _analyticsService = analyticsService;
        }

        [HttpGet("orders-with-users")]
        public async Task<IActionResult> GetOrdersWithUsers([FromQuery] int limit = 10, [FromQuery] int offset = 0)
        {
            try
            {
                var orders = await _analyticsService.GetOrdersWithUsersAsync(limit, offset);
                return Ok(orders);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpGet("user-order-summary")]
        public async Task<IActionResult> GetUserOrderSummary([FromQuery] int limit = 10, [FromQuery] int offset = 0)
        {
            try
            {
                var summaries = await _analyticsService.GetUserOrderSummaryAsync(limit, offset);
                return Ok(summaries);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpGet("analytics")]
        public async Task<IActionResult> GetComplexAnalytics()
        {
            try
            {
                var analytics = await _analyticsService.GetComplexAnalyticsAsync();
                return Ok(analytics);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpGet("health")]
        public IActionResult HealthCheck()
        {
            return Ok(new
            {
                status = "healthy",
                timestamp = DateTime.UtcNow,
                service = "dotnet-api"
            });
        }
    }
} 